src/api/handlers/settlements/{id}.js

```diff
@@ -73,6 +74,7 @@ module.exports = {
       if (p.participants && (p.state || p.reason || p.externalReference)) {
         throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.VALIDATION_ERROR, 'No other properties are allowed when participants is provided')
       } else if ((p.state && !p.reason) || (!p.state && p.reason)) {
+        Logger.isErrorEnabled && Logger.error('State and reason are mandatory')
```

Comment:
I would prefer that the error is created, and then printed using the logger before it is thrown.

The reason for this being that we should get a stack trace being printed to the lock which includes the line of where the error happened etc.

Let me know your thoughts here.

Reply:
> I would prefer that the error is created, and then printed using the logger before it is thrown.

done.

> Something like this --> Logger.throwError(Error err)

Would be a nice to have in the future, save some lines and keystrokes. I feel neutral on it. I can create a feature request for `central-services-shared` if you like that idea.

Reply:
> Would be a nice to have in the future, save some lines and keystrokes. I feel neutral on it. I can create a feature request for `central-services-shared` if you like that idea.

Yeah agreed. 

It would make things a bit more cleaner...

`Logger.errorWithThrow` <-- perhaps that works better than `Logger.throwError`.